base/dates/io.jl

```diff
@@ -152,7 +152,7 @@ function parse(x::AbstractString,df::DateFormat)
         throw(ArgumentError("Delimiter mismatch. Couldn't find first delimiter, \"$(df.slots[1].transition)\", in date string"))
     end
     periods = Period[]
-    extra = []
+    extra = Period[]
```

Comment:
Here is an example of why `extra` needs to be included:

```julia
julia> using TimeZones

julia> Dates.parse("1942-12-25T01:23:45+0100", Dates.DateFormat("yyyy-mm-ddTHH:MM:SSzzz", "english"))
7-element Array{Any,1}:
 1942 years                                                                           
 12 months                                                                            
 25 days                                                                              
 1 hour                                                                               
 23 minutes                                                                           
 45 seconds                                                                           
 TimeZones.FixedTimeZone(symbol("UTC+01:00"),TimeZones.Offset(3600 seconds,0 seconds))
```

Excluding extra code:
```julia
julia> using TimeZones

julia> Dates.parse("1942-12-25T01:23:45+0100", Dates.DateFormat("yyyy-mm-ddTHH:MM:SSzzz", "english"))
ERROR: MethodError: `convert` has no method matching convert(::Type{Base.Dates.Period}, ::TimeZones.FixedTimeZone)
This may have arisen from a call to the constructor Base.Dates.Period(...),
since type constructors fall back to convert methods.
Closest candidates are:
  call{T}(::Type{T}, ::Any)
  convert{T<:Base.Dates.Period}(::Type{T<:Base.Dates.Period}, ::Real)
  convert{T}(::Type{T}, ::T)
 in push! at array.jl:422
 [inlined code] from bool.jl:16
 in parse at dates/io.jl:158
```

Since TimeZones.jl is currently (as far as I know) the only package extending the slots which means `extra` will only be of type `TimeZone` but since we don't know how this parser could be extended in the future I think we should make the extra definition as explicit as possible:

```julia
extra = Any[]  # Supports custom slot types such as TimeZone
```